src/components/InspirationSection.vue

```diff
@@ -53,65 +53,49 @@
   </section>
 </template>
 
-<script>
+<script setup lang="ts">
+import { ref, computed, onMounted } from 'vue'
 import TheButton from './shared/TheButton.vue'
 import TheWorkout from './shared/TheWorkout.vue'
 import TheExercise from './shared/TheExercise.vue'
 import { supabase } from '../supabase'
+import { Workout, WorkoutResponse } from '../api/types'
 
-export default {
-  name: 'InspirationSection',
-  components: {
-    TheButton,
-    TheWorkout,
-    TheExercise
-  },
-  data() {
-    return {
-      workouts: [],
-      isError: false,
-      workoutNumber: 0
-    }
-  },
-  computed: {
-    message() {
-      return this.isError ? 'Ooops, something went wrong when fetching data. Try refreshing the page.' : 'Take a look at other’s workouts below!'
-    },
-    currentWorkout() {
-      return this.workouts.filter((workout) => workout === this.workouts[this.workoutNumber])
-    }
-  },
-  methods: {
-    increaseWorkoutNumber() {
-      this.workoutNumber += 1
-
-      if (!this.workouts[this.workoutNumber]) {
-        this.workoutNumber = 0
-      }
-    },
-    decreaseWorkoutNumber() {
-      this.workoutNumber -= 1
-
-      if (!this.workouts[this.workoutNumber]) {
-        this.workoutNumber = this.workouts.length - 1
-      }
-    }
-  },
-  async mounted() {
-    const fetchWorkouts = async () => {
-      const response = await supabase.from('Workouts').select()
-      if (response.error === null) {
-        const { data } = response
-
-        this.workouts = data.map(({ workout: { name, exercises } }) => ({ name, exercises }))
-
-        this.isError = false
-      } else {
-        this.isError = true
-      }
-    }
-
-    await fetchWorkouts()
+const workouts = ref<Workout[]>([])
+const isError = ref(false)
+const message = computed(() => isError.value ? 'Ooops, something went wrong when fetching data. Try refreshing the page.' : 'Take a look at other’s workouts below!')
+
+const fetchWorkouts = async () => {
+  const response = await supabase.from('Workouts').select()
```

Comment:
I had thought about making it its own interface previously but ultimately I decided it would look weird having this interface for the response from a database.

However, since you think it would be actually nice 😛, I **added it**. Nevertheless, I'm not sure I handled this right (as I mentioned many times before I really suck in writing this TS thing 🙈

I had some issues in my current approach, which I'd like to draw your attention to:
1) The response actually varies depending on the error being or not being returned - I built my interface on two responses - one without error and the other one with error caused by giving a wrong table name in the related function. However, I didn't test other errors, so I don't know if this response won't change for them, e.g. the `count` property being something else than `null`, since it's my understanding this property was put there for some reason and if it always was `null` what's the purpose of its existence?
2) In the second line of this: 
```js
const response: WorkoutsResponse = await supabase.from('Workouts').select()
const { data }: { data: WorkoutsResponse['data'] } = response
```
I had previoulsy another type pointing the array of workouts (I changed the name for this interface so I suppose this time it'll be easier for you if I just explain instead of giving names) but when I typed the response (first line) as `WorkoutsResponse` I got this error on `data` that the type I provided previously is not assignable to the current type of `WorkoutsResponse` (or something like that, don't remember well). So I thought, maybe I was able to use `data` property inside the `WorkoutsResponse` as a type. At first, I tried to write like so: `WorkoutsResponse.data` but then I got another error saying basically I couldn't do that for some reason I didn't understand well and if I had meant to write it like it is currently written? So I did it this way and it seems to work but I have no idea what I actually did and why this is working 😛

Reply:
Sorry but I didn't understand. Can you provide an example?

Reply:
1. I assume the response object may vary depending for example on the error type and if it's different than the two (response objects) I saw (one with error and one without it), my interface won't cover it. Below, you can see those response objects for the mentioned scenarios - as it can be seen, there are already some differences between them and I have no way of knowing if those two are the only possible ones:
![error](https://user-images.githubusercontent.com/113309294/228045022-c52cb4aa-5b41-4af3-a23b-9750ccb52277.PNG)
![no-error](https://user-images.githubusercontent.com/113309294/228045026-79129e0e-dc36-43ff-9fd9-33996e33a9e9.PNG)

2. I just want to know if the cited lines of code are okey... I mean, my types work (have no errors from TS) but I had been just trying different things until one worked so I'm not sure if this way of typing of destructured properties is right.

Reply:
1. Error is optional and might be null, that's all to it
```ts
interface Example {
  requiredProp: 1;
  optionalProp?: 2;
}
```
2. Cited looks better now, but there's still room for improvement (I mean the `WholeType` thing, which I've commented in another thread.

Reply:
I'm sorry but I don't know what you mean by this `WholeType` thing.

Also, error isn't optional, it's either an error object or null, but it's always there.